From: Tomas Henzl <thenzl@redhat.com>
Date: Fri, 25 Feb 2011 14:57:18 -0500
Subject: [message] mptfusion: add ioc_reset_in_progress reset in SoftReset
Message-id: <1298645850-4583-8-git-send-email-thenzl@redhat.com>
Patchwork-id: 33630
O-Subject: [RHEL5.7 PATCH 07/19] mptfusion: Added missing reset for
	ioc_reset_in_progress in SoftReset
Bugzilla: 662160

Added missing part which will reset ioc_reset_in_progress before returning from SoftResetHandler.


diff --git a/drivers/message/fusion/mptbase.c b/drivers/message/fusion/mptbase.c
index 95d3d4f..9b51bf6 100644
--- a/drivers/message/fusion/mptbase.c
+++ b/drivers/message/fusion/mptbase.c
@@ -6982,6 +6982,7 @@ mpt_SoftResetHandler(MPT_ADAPTER *ioc, int sleepFlag)
 
 	spin_lock_irqsave(&ioc->taskmgmt_lock, flags);
 	if (ioc->taskmgmt_in_progress) {
+		ioc->ioc_reset_in_progress = 0;
 		spin_unlock_irqrestore(&ioc->taskmgmt_lock, flags);
 		return -1;
 	}
